### PR TITLE
feat: use user login as user identity

### PR DESCRIPTION
**Pull request by gabrielfs7**
https://oat-sa.atlassian.net/browse/TR-4941

# Goal

Use user login instead or URI as identifier, so it can later be user by data store + tao insights

# How to test

- Launch a test from CG using LTI 1.3 and change the token contents

# Example how it worked: 

OIDC token

`
eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiIsImtpZCI6IjFfMCJ9.eyJodHRwczovL3B1cmwuaW1zZ2xvYmFsLm9yZy9zcGVjL2x0aS9jbGFpbS9yZXNvdXJjZV9saW5rIjp7ImlkIjoiYTc1YzllNmJhNjE5In0sImh0dHBzOi8vcHVybC5pbXNnbG9iYWwub3JnL3NwZWMvbHRpL2NsYWltL3ZlcnNpb24iOiIxLjMuMCIsImh0dHBzOi8vcHVybC5pbXNnbG9iYWwub3JnL3NwZWMvbHRpL2NsYWltL21lc3NhZ2VfdHlwZSI6Ikx0aVJlc291cmNlTGlua1JlcXVlc3QiLCJodHRwczovL3B1cmwuaW1zZ2xvYmFsLm9yZy9zcGVjL2x0aS9jbGFpbS9kZXBsb3ltZW50X2lkIjoiMSIsImh0dHBzOi8vcHVybC5pbXNnbG9iYWwub3JnL3NwZWMvbHRpL2NsYWltL3RhcmdldF9saW5rX3VyaSI6Imh0dHBzOi8vZGVsaXZlci5kb2NrZXIubG9jYWxob3N0L2FwaS92MS9hdXRoL2xhdW5jaC1sdGktMXAzL2E3NWM5ZTZiYTYxOSIsImh0dHBzOi8vcHVybC5pbXNnbG9iYWwub3JnL3NwZWMvbHRpL2NsYWltL3JvbGVzIjpbIkxlYXJuZXIiXSwicmVnaXN0cmF0aW9uX2lkIjoiMV8wIiwiaHR0cHM6Ly9wdXJsLmltc2dsb2JhbC5vcmcvc3BlYy9sdGktYm8vY2xhaW0vYmFzaWNvdXRjb21lIjp7Imxpc19yZXN1bHRfc291cmNlZGlkIjoiaHR0cDovL2JhY2tvZmZpY2UuZG9ja2VyLmxvY2FsaG9zdC9vbnRvbG9naWVzL3Rhby5yZGYjaTYzYjU4ODIyYTY5YjgxMzAzYTM3YzZlZDI5ODk3YWU2MiIsImxpc19vdXRjb21lX3NlcnZpY2VfdXJsIjoiaHR0cDovL2JhY2tvZmZpY2UuZG9ja2VyLmxvY2FsaG9zdC90YW9MdGlDb25zdW1lci9SZXN1bHRDb250cm9sbGVyL21hbmFnZVJlc3VsdHMifSwibm9uY2UiOiJjOTg3N2Q5Zi1jYjRkLTQwMWQtYmU2Zi01MWNjZmQyMTc3YTYiLCJqdGkiOiJjOTI3MjBhYS0zNGEzLTRjZmEtOGNmNy1jYjQ4ZmFiZGE5ZmQiLCJpYXQiOjE2NzI5MDU0MTMuMTc0MDczLCJuYmYiOjE2NzI5MDU0MTMuMTc0MDczLCJleHAiOjE2NzI5MDYwMTMuMTc0MDczLCJpc3MiOiJodHRwOi8vYmFja29mZmljZS5kb2NrZXIubG9jYWxob3N0IiwiYXVkIjoiYmFja29mZmljZS1kZWxpdmVyLWlkIiwic3ViIjoiZ2FicmllbCIsIm5hbWUiOiJHYWJyaWVsIFNvYXJlcyIsImVtYWlsIjoiZ2FicmllbC5mZWxpcGUuc29hcmVzQHRhb3Rlc3RpbmcuY29tIn0.Rksf5CATsRQOP8Db4vu_GhakXjhpc0nzVrKkvuuKgxRuzwa5CYQsw6BfO3q7jBvRHTHqzsMhEi00LsTLNHgnNGdtGwI0CRqYa6u8MQI95lbt4nIT3QuR_QY-cCWAP08dhReSbDhMtl3ybndBwrsteE6JP52KJqkgaSkt4QjHLR54NsuvjrCIShG4I855QdRRqqrxRZRA5s9gxq7-tJt_bYJsoLQKflz3cxy_hzeKhqQFcsbpkFrIDhM21FxROsKBEZPdFXhvnZ_pYreiJadUk9CmxeszLdM_tOIz2Ewo3L_X3BkfbpJthUxc3-H8m1yfABo0XbC56lZhdC12DVgLjA
`

Screenshot 2023-01-09 at 14 26 25

### CODE REVIEW

File: models/classes/Platform/Service/Oidc/Lti1p3UserAuthenticator.php [hunk 1]
```diff
@@ -57,12 +61,21 @@ private function getUserIdentity(string $userId): UserIdentity
             throw new ErrorException(sprintf('User [%s] not found', $userId));
         }
 
-        $fullName = (string)($user->getPropertyValues(GenerisRdf::PROPERTY_USER_FIRSTNAME)[0] ?? null)
-            . ' ' . (string)($user->getPropertyValues(GenerisRdf::PROPERTY_USER_LASTNAME)[0] ?? null);
+        $login = $user instanceof AnonymousUser || $user instanceof GuestTestUser ?
+            $userId :
+            $this->getPropertyValue($user, UserRdf::PROPERTY_LOGIN);
```
Comment (shpran) [inline models/classes/Platform/Service/Oidc/Lti1p3UserAuthenticator.php:66]:
```suggestion
        $login = $user instanceof AnonymousUser || $user instanceof GuestTestUser
            ? $userId
            : $this->getPropertyValue($user, UserRdf::PROPERTY_LOGIN);
```

**Review by shpran — APPROVED** [review]
- [x] New code is covered by tests (if applicable)
- [x] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [ ] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful
- [x] Pull request's target is not `master`
- [x] Commits are following [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] Commits messages are meaningful
- [x] Commits are atomic
- [x] Changelog is updated according to changes (if applicable)
- [x] Documentation is updated according to changes (if applicable)

**Review by hectoras — APPROVED** [review]
- [x] Pull request title and description are meaningful
- [x] Pull request's target is not `master`
- [x] Target branch is correct
- [x] Commits are following conventional commits
- [x] New code is covered by tests (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Kitchen links are not exposed in the description

**Review by rgomeztao — APPROVED** [review]
nice job! :)

**Pull request merged**